test/Test/Data/List/Lazy.purs

```diff
@@ -332,8 +334,15 @@ testListLazy = do
   assert $ nub (l [1, 2, 2, 3, 4, 1]) == l [1, 2, 3, 4]
 
   log "nubBy should remove duplicate items from the list using a supplied predicate"
-  let nubPred = \x y -> if odd x then false else x == y
-  assert $ nubBy nubPred (l [1, 2, 2, 3, 3, 4, 4, 1]) == l [1, 2, 3, 3, 4, 1]
+  let nubPred = compare `on` Array.length
```

Comment:
Could you add a test which ensures that nub behaves sensibly on infinite lists please? Perhaps
```
log "nub should not consume more of the input list than necessary"
assert $ (take 3 $ nub $ cycle $ l [1,2,3]) == l [1,2,3]
```

Reply:
Should this test be applied to all `nub*` functions, or just `nub`? I just added the single test.

Reply:
I think it’s worth doing it for nub and nubEq but we probably don’t need it for the *By variants, as if it works for nub it’ll probably work for nubBy too.